packages/admin-panel/src/table/DataFetchingTable/Pagination.jsx

```diff
@@ -0,0 +1,176 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { IconButton, Input, Typography } from '@material-ui/core';
+import styled from 'styled-components';
+import { KeyboardArrowLeft, KeyboardArrowRight } from '@material-ui/icons';
+import { Select } from '@tupaia/ui-components';
+
+const Wrapper = styled.div`
+  margin-block: 0.9rem;
+  font-size: 0.75rem;
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  label,
+  p,
+  .MuiInputBase-input {
+    font-size: 0.75rem;
+  }
+`;
+
+const ActionsWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  :first-child {
+    padding-inline-start: 1rem;
+  }
+`;
+const Button = styled(IconButton)`
+  border: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  padding: 0.4rem;
+  .MuiSvgIcon-root {
+    font-size: 1.2rem;
+  }
+  & + & {
+    margin-left: 0.7rem;
+  }
+`;
+
+const ManualPageInputContainer = styled.div`
+  display: flex;
+  align-items: center;
+  margin-inline-start: 0.5rem;
+  margin-inline-end: 0.8rem;
+`;
+
+const ManualPageInput = styled(Input)`
+  border: 1px solid ${({ theme }) => theme.palette.grey['400']};
+  border-radius: 4px;
+  padding-block: 0.5rem;
+  padding-inline: 0.8rem 0.2rem;
+  margin-inline: 0.5rem;
+  font-size: 0.75rem;
+  .MuiInputBase-input {
+    text-align: center;
+    padding-block: 0;
+    height: auto;
+  }
+`;
+
+const Text = styled(Typography)`
+  font-size: 0.75rem;
+`;
+
+const RowsSelect = styled(Select)`
+  display: flex;
+  flex-direction: row;
+  margin-block-end: 0;
+  margin-inline-end: 1.2rem;
+  width: 12rem;
+  .MuiSelect-root {
+    padding-block: 0.5rem;
+    padding-inline: 0.8rem 0.2rem;
+  }
+`;
+
+const PageSelectComponent = ({ onChangePage, page, pageCount }) => {
+  const pageDisplay = page + 1;
+  return (
+    <ActionsWrapper>
+      <ManualPageInputContainer>
+        <Text id="page-label" htmlFor="page" component="label">
+          Page
+        </Text>
+        <ManualPageInput
+          type="number"
+          value={pageDisplay}
+          onChange={e => {
+            const newPage = e.target.value ? Number(e.target.value) - 1 : '';
```

Comment:
Don’t know the ins-and-outs of _why_, but using the `Number` constructor [seems to be discouraged](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/Number)

```suggestion
            const newPage = e.target.value ? Number.parseInt(e.target.value) - 1 : '';
```